File: app/Models/UserActivity.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class UserActivity extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'browser_name',
        'platform',
        'device',
        'ip',
        'clicked_url',
    ];

    // protected $hidden = [
    //     'ip',
    // ];
}

File: src/models/user_activity.go
package models

type UserActivity struct {
	DatabaseFields
	UserID      uint   `gorm:"not null" json:"user_id"`
	BrowserName string `gorm:"size 255;default:unknown" json:"browser_name"`
	Platform    string `gorm:"size 255;default:unknown" json:"platform"`
	Device      string `gorm:"size 255;default:unknown" json:"device"`
	Ip          string `gorm:"size 255;default:unknown" json:"ip"`
	ClickedUrl  string `gorm:"size 255;default:unknown" json:"clicked_url"`

	User *User `gorm:"foreignKey:UserID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;" json:"-"`
}

func (ua UserActivity) TableName() string {
	return "user_activities"
}
